backend/sftp/fileSystem.go

```diff
@@ -89,24 +95,55 @@ func (fs *FileSystem) Scheme() string {
 // Client returns the underlying sftp client, creating it, if necessary
 // See Overview for authentication resolution
 func (fs *FileSystem) Client(authority utils.Authority) (Client, error) {
+	// first stop connection timer, if any
+	fs.connTimerStop()
 	if fs.sftpclient == nil {
 		if fs.options == nil {
 			fs.options = Options{}
 		}
 
-		if opts, ok := fs.options.(Options); ok {
-			var err error
-			fs.sftpclient, err = fs.getClient(authority, opts)
-			if err != nil {
-				return nil, err
-			}
-		} else {
+		opts, ok := fs.options.(Options)
+		if !ok {
 			return nil, fmt.Errorf("unable to create client, vfs.Options must be an sftp.Options")
 		}
+		var err error
+		fs.sftpclient, err = defaultClientGetter(authority, opts)
+		if err != nil {
+			return nil, err
+		}
 	}
 	return fs.sftpclient, nil
 }
 
+func (fs *FileSystem) connTimerStart() {
+	fs.timerMutex.Lock()
+	defer fs.timerMutex.Unlock()
+
+	aliveSec := defaultAutoDisconnectDuration
+	if fs.options != nil {
+		if v, ok := fs.options.(Options); ok && v.AutoDisconnect != 0 {
+			aliveSec = v.AutoDisconnect
+		}
+	}
+
+	fs.connTimer = time.AfterFunc(time.Duration(aliveSec)*time.Second, func() {
+		// close connection and nil-ify client to force lazy reconnect
+		if fs.sftpclient != nil {
+			_ = fs.sftpclient.Close()
+			fs.sftpclient = nil
+		}
+	})
```

Comment:
This ends up being the meat of the changes.  This timer will disconnect after 10 seconds (default) unless a new request to the server is made (read, write, list, etc), in which case the timer will reset.  
If a timer expires and closes a connection, any subsequent server request action will request a new client, reconnecting and starting new timer.
In this way, a user need not explicit call Close on a client since it will simply timeout anyway.